### PR TITLE
Espresso tests rework

### PR DESCRIPTION
**Pull request by tobrun**
This PR does a cleanup of our instrumentation tests by making a distinction between tests that: 
 - load a style through the hosting activity (eg. activity sanity tests)
 - load a style through the test (eg. testing business logic on top of style)

Notable changes:
 - moved activity based tests to instrumented unit tests if possible
 - removed the SingleActivity/SingleFragment setup migrated with LocationComponent
 - performs cleanup of unused methods/classes
 - remove ignored tests that have been ignored for a while

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/activity.junit.ejs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class <%- activity %>Test extends BaseActivityTest {
 
     @Test
     public void testSanity() {
-        validateTestSetup();
+        // no-op
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/activity.junit.ejs]: Can we remove this test all-together?

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/utils/FinishLoadingStyleIdlingResource.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  @WorkerThread
+  public FinishLoadingStyleIdlingResource(final Activity activity) {
+    new Handler(Looper.getMainLooper()).post(() -> inflateMap(activity));
+  }
+
+  @Override
+  public void initMap(MapboxMap mapboxMap) {
+    super.initMap(mapboxMap);
+    mapboxMap.getStyle(style -> {
+      if (resourceCallback != null) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/utils/FinishLoadingStyleIdlingResource.java]: Can we add an assertion checking that the style is fully loaded?

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/utils/LoadStyleIdlingResource.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  @WorkerThread
+  public LoadStyleIdlingResource(final Activity activity) {
+    new Handler(Looper.getMainLooper()).post(() -> inflateMap(activity));
+  }
+
+  @UiThread
+  public void initMap(MapboxMap mapboxMap) {
+    super.initMap(mapboxMap);
+    mapboxMap.setStyle("asset://streets.json", style -> {
+      this.style = style;
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/utils/LoadStyleIdlingResource.java]: Same as above.

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/EspressoActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import com.mapbox.mapboxsdk.testapp.activity.espresso.EspressoTestActivity;
+import com.mapbox.mapboxsdk.testapp.utils.LoadStyleIdlingResource;
+
+/**
+ * Base class for all tests using EspressoTestActivity as wrapper.
+ * <p>
+ * Uses {@link LoadStyleIdlingResource} to load "assets/streets.json" as style.
+ * </p>
+ */
+public class EspressoActivityTest extends BaseActivityTest {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/EspressoActivityTest.java]: Can we rename `EspressoActivityTest` to `EspressoTest`? Similarly with `BaseActivityTest`. It's a little bit hard to read and distinguish between `EspressoTestActivity`.

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/activity.junit.ejs (hunk 1 above)
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/activity.junit.ejs]: For now I would keep the sanity tests around, updated above block however with assertions for mapboxMap, style and fully loaded.

**Pull request merged**